### PR TITLE
remark-captions: fix captions for images

### PR DESCRIPTION
**Pull request by AmarOk1412**
cf #95

### CODE REVIEW

File: packages/remark-captions/__tests__/index.js [hunk 1]
```diff
@@ -212,3 +212,38 @@ test('internal legend: two legends', () => {
...
+    aFigure: 2 this is displayed as text
+
+    ![alt 2b](https://zestedesavoir.com/static/images/home-clem.4a2f792744c9.png)
+    this is displayed as text
+
+    foo
+    ![]()
+    Figure: 3 this is a legend
+
+    ![]()
```
Comment (artragis) [inline packages/remark-captions/__tests__/index.js]: you forgot a similar case that is really usefull to handle : 

```markdown

![]()
Figure: the true figure

Figure: this one must stay as is.

File: packages/remark-captions/dist/index.js [hunk 2]
```diff
@@ -35,8 +35,8 @@ function internLegendVisitor(internalBlocks) {
     if (parent && parent.type === 'figure') return;
 
     // legend can only be in a paragraph
-    var lastP = getLast(node.children);
-    if (!lastP || lastP.type !== 'paragraph') return;
+    var lastP = node.children ? getLast(node.children) : parent;
+    if (!lastP || node.children && lastP.type !== 'paragraph' || !node.children && parent.type !== 'paragraph') return;
```
Comment (artragis) [inline packages/remark-captions/dist/index.js]: can you put a comment as a "4 conditional assertion with use of priority" is not the easyest thing to read.

File: packages/remark-captions/__tests__/index.js (hunk 1 above)
Comment (AmarOk1412) [inline packages/remark-captions/__tests__/index.js]: @artragis updated!

**Comment by artragis** [comment]
amha it's mergeable.

**Pull request merged**